x/evm/keeper/grpc_user_smart_contracts.go

```diff
@@ -0,0 +1,27 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/palomachain/paloma/x/evm/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) GetUserSmartContracts(
+	ctx context.Context,
+	req *types.QueryUserSmartContractsRequest,
+) (*types.QueryUserSmartContractsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	contracts, err := k.UserSmartContracts(ctx, req.ValAddress)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You might want to catch and handle `keeperutil.ErrNotFound`. I always find the message somewhat too cryptic to return to the CLI. But that's just me.

Reply:
In this specific case, though, it returns a list using `keeperutil.IterAll`, which will return an empty slice instead of `ErrNotFound`, so it should be fine here.